core/src/main/scala/com/karumi/shot/screenshots/ScreenshotsComparator.scala

```diff
@@ -0,0 +1,61 @@
+package com.karumi.shot.screenshots
+
+import java.io.File
+
+import com.karumi.shot.domain._
+import com.karumi.shot.domain.model.ScreenshotsSuite
+import com.sksamuel.scrimage.{AwtImage, Color, Image}
+
+object ScreenshotsComparator {
+  private val tileSize = 512
+}
+
+class ScreenshotsComparator {
+
+  import ScreenshotsComparator._
+
+  def compare(screenshots: ScreenshotsSuite): ScreenshotsComparisionResult = {
+    val errors = screenshots.par.flatMap(compareScreenshot).toList
+    ScreenshotsComparisionResult(errors, screenshots)
+  }
+
+  private def compareScreenshot(
+      screenshot: Screenshot): Option[ScreenshotComparisionError] = {
```

Comment:
Cool, I've been struggling a bit with the return type of operations like this one, where if everything goes ok you don't want to return anything but in case of an error you want to know what happened, the more you know!